### DISCUSSION TITLE
A mail app, to manage mail server from docker

### DISCUSSION THREAD

**Discussion by miguelgargallo** [coollabsio/coolify#1142] [Service Template Requests]
This would solve startups email, with Mail apps, there is none

**Comment by Jaffa-Cakes** [comment]
It would be great if we could get something fully featured like Mailcow or docker-mailserver :)

**Reply by MelchiorKokernoot** [reply]
I have docker mailserver running, was pretty alright to setup just following the basic instructions. Only thing i am having some trouble with is TLS configuration. I just created a package repo where I have the .env and docker compose files configured.

**Comment by rodrgds** [comment]
I feel like https://github.com/Mailu/Mailu might be the easiest since it's Docker based? Just an idea - would be cool to have a template for it.

**Reply by miguelgargallo** [reply]
lemme see

**Comment by gitekDev** [comment]
+1 for [Mailcow](https://github.com/mailcow/mailcow-dockerized)

**Comment by ItzAmirreza** [comment]
Looking forward to this! It's a serious feature that needs to be implemented.

**Reply by yes-oo** [reply]
i agree !

**Comment by ooomytech** [comment]
[Stalwart Mail Server](https://github.com/stalwartlabs/mail-server) - As an alternative, it can also be used.
Overall, it can already be used right now.

**Reply by ItzAmirreza** [reply]
> [Stalwart Mail Server](https://github.com/stalwartlabs/mail-server) - As an alternative, it can also be used.
> 
> Overall, it can already be used right now.
> 
> 


By any chance can you share the setup config for coolify? I have tried mailu but I seem to have problems with SSL certification and traefik.

**Reply by cntrvsy** [reply]
I second this, im currently going through the same issue with SSL certificate with traefik making the site not accessible via https (stalwarts/mail server). Also if you can link any resources on using it outside of the document that will be helpful.

**Reply by MelchiorKokernoot** [reply]
You can easily fix this by adding a whoami container to your docker-compose file:
```
whoami:
    image: 'docker.io/traefik/whoami:latest'
```

**Comment by ooomytech** [comment]
Create a Docker Compose-based application. I provided an example of how I use it. For HTTPS to work and to proxy traffic, you need to specify the domain along with the port. After that, you can enter the panel address in the browser without explicitly specifying the port.

![image](https://github.com/user-attachments/assets/acdc74b1-1bd3-423e-813f-3518d94f406e)

Additionally, after creating your application, you need to check the logs where you’ll see the login and password for the administrator. I manually generated the TLS certificates for the server itself, but there is an option to generate them automatically, though I encountered some issues with that. Overall, the server works fine, and it sends and receives mail.


Config  for [Stalwart Mail Server](https://github.com/stalwartlabs/mail-server)

Docker Compose:

```version: '3.8'
services:
  stalwart-mail:
    image: 'stalwartlabs/mail-server:latest'
    container_name: stalwart-mail
    ports:            
      - '25:25'
      - '587:587'
      - '465:465'
      - '143:143'
      - '993:993'
      - '4190:4190'
      - '110:110'
      - '995:995'
    volumes:
      - '<STALWART_DIR>:/opt/stalwart-mail'
    tty: true
    stdin_open: true
    restart: always
```

**Reply by innocent-greats** [reply]
can you assist,  i have successfuly deployed but i am not gettin credentials, trying to run docker logs stalwart-mail and its giving me 

![Screenshot 2024-10-11 102320](https://github.com/user-attachments/assets/73aa8925-32f3-4cb1-b042-43326c4b1f12)

**Reply by CareProID** [reply]
> can you assist, i have successfuly deployed but i am not gettin credentials, trying to run docker logs stalwart-mail and its giving me
> 
> ![Screenshot 2024-10-11 102320](https://private-user-images.githubusercontent.com/97280792/375682245-73aa8925-32f3-4cb1-b042-43326c4b1f12.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjkyMTg2NTYsIm5iZiI6MTcyOTIxODM1NiwicGF0aCI6Ii85NzI4MDc5Mi8zNzU2ODIyNDUtNzNhYTg5MjUtMzJmMy00Y2IxLWIwNDItNDMzMjZjNGIxZjEyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDE4VDAyMjU1NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhMDBmMDI3MDdiN2FhNGJkZGJhYzE1OTFjMzkyOTk2YmFiNTIwZTczYTY3ODcwMzMwODFlZWMwOWU3YTQ0MDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mLIz4CoZg4ZrHESAqrLXklV89CGIEXRDpKeBYylD3-o)

You should see on Application logs when the container is run for the first time.

![Screenshot 2024-10-18 at 09 28 21](https://github.com/user-attachments/assets/895f0552-cd2e-42f1-89f8-d5267e56e94f)

**Reply by antonind72** [reply]
how do you generate the ssl/tls certificate? and where do you manage your dns? can you show the steps to generate the certifica? please

**Reply by Lqm1** [reply]
I am also curious about how to set ssl/tls.
I have set it up, but I don't know how to set up ssl/tls

**Reply by invays** [reply]
Hi! Idk why, but after status - Started, I have no logs info... Only message that I need to restart the page. Mb i did something wrong ?

**Reply by phykas** [reply]
<STALWART_DIR> needs to be created and referenced from host machine in order to get username and password in logs.
example that works:
    volumes:
      - '/var/lib/stalwart-mail:/opt/stalwart-mail'

**Reply by ricardoabr** [reply]
Hi @ooomytech !
I´ve installed stalwart following you directions, tks. I used the option "notls" because my server is on a Cloudflare tunnel, already https only. Did not finish the configs but its going well so far. Something tricky was that I coud get the login credentials in the logs only if I had no port forwarding, so I reinstalled, toke note of the credentials and then, updated the compose file with '9000:8080' and restarted the service. Wanted to post it here because maybe that can help someone.

**Comment by yborisovp-old** [comment]
+1

**Comment by cntrvsy** [comment]
@ooomytech, sorry to bother you but i cannot for the life of me figure out what's wrong with my deployment, I have been able to get the mail server running on my vps and i can send emails but for some reason i cannot receive emails ( using mail-tester.com i receive a score of 8/10 missing out on 2 point as i have more than one DKIM signature in your message.), did some digging and my connection to the SMTP server is successful, but the issue is with the certificate verification returning the error code: 18 (self-signed certificate) which im assuming isn't trusted by default (i didn't think this would be a problem as i thought coolify does the SSL Certificate for me), I'm pretty new to mail servers and VPS and all the information is general overwhelming. Im doing all of this in the hopes of fixing another error as i would my pocketbase instance running on the same vps to be able to use my custom emails but i get the error "Failed to send the test email. Raw error: tls: failed to verify certificate: x509: certificate is valid for localhost, not mail.company.com". any guidance will be appreciated.

**Reply by ooomytech** [reply]
use the command: certbot certonly --manual --preferred-challenges dns -d yourdomain.com -d *.yourdomain.com.

**Comment by ooomytech** [comment]
To create certificates using Certbot with DNS verification, use the command: certbot certonly --manual --preferred-challenges dns -d yourdomain.com -d *.yourdomain.com. Verify the process by adding DNS records. Then upload the obtained certificates (fullchain.pem and privkey.pem) in the web interface in the TLS section (Stalwart)


Make sure that the ports for sending mail are open. On Hetzner, ports may be blocked for new accounts, so you need to send a request to support to unblock them.

**Reply by antonind72** [reply]
my site currently uses mydomain.com with cloudflare in ssl certifica, will it block me if I want an email contact@mydomain.com?

or do I just need to generate a certifica for mail.mydomain.com?

**Reply by cntrvsy** [reply]
for the portion "Then upload the obtained certificates (fullchain.pem and privkey.pem) in the web interface in the SSL section." which web interface are we talking about?

**Reply by antonind72** [reply]
> for the portion "Then upload the obtained certificates (fullchain.pem and privkey.pem) in the web interface in the SSL section." whic

I think he's talking about stalwart

**Reply by cntrvsy** [reply]
@antonind72 i for the life of me cannot find the SSL section on the web interface for stalwart

**Reply by ooomytech** [reply]
![image](https://github.com/user-attachments/assets/8e157a8b-2326-4199-b089-1979b84c96af)

@antonind72 i for the life of me cannot find the SSL section on the web interface for stalwart

**Reply by cntrvsy** [reply]
thanks for everything, everything working now as expected

**Reply by PAYDAYTEAM** [reply]
Hey! Thanks for the help, you are a god, I was able to setup everything and is working as expected but it keeps going to spam folder, is there any config on DKIM or SPF to at least try to lower the percentage of mails flagged as spam?

**Reply by PAYDAYTEAM** [reply]
Already solved it! I was able to stop the spam flag writing all the DNS records in the DNS zone of my domain including TXT, TLSA and CNAME (for easyconfig on devices). Then I made a small change in
 the SPF TXT to refer directly to the MX record. Example of my modified value: TXT | yourdomain.com | "v=spf1 mx a -all"

PS: Still happening in certain cases like personal gmail accounts but working for other corporate emails.

**Reply by TMBL-DEV** [reply]
i just get a tls error any clue on how to fix that??

**Comment by mredodos** [comment]
+1 for mailcow

**Reply by cntrvsy** [reply]
honestly looking back i should have tried mailcow first instead of being cutting edge and but you live and you learn i guess.

**Reply by stevebrainng** [reply]
> honestly looking back i should have tried mailcow first instead of being cutting edge and but you live and you learn i guess.

Do you have any particular reason for that? What features would you have loved to use Mailcow for that this (stalwart) doesn't already have.

Any comment please??

**Comment by dr-diffie** [comment]
+1 for Stalwart. Was mentioned here as well https://github.com/coollabsio/coolify/discussions/4118

**Comment by dr-diffie** [comment]
@ysfaran

**Comment by roo12312** [comment]
need this

**Comment by ApoloApps** [comment]
need this!!!

**Comment by xer0xde** [comment]
+1 for Mailcow

**Comment by Tunahan-Uysal** [comment]
would be great

**Comment by Tebra** [comment]
Is there any news on this yet? Thanks for the great tool! Love using it ❤️

**Comment by jmischler72** [comment]
Hello, if anyone wants to still use stalwart with the docker compose file provided by @ooomytech, you need to change the docker image to `stalwartlabs/stalwart:latest` since `stalwartlabs/mail-server:latest` only goes to v0.11,  i had an issue where my web client told me that the server used an older version

**Comment by ThomasProgli** [comment]
+1 for stalwart

**Reply by miguelgargallo** [reply]
I checked stalwart, on the roadmap is "Mail client", lets be patient, I like what I see by the way, let me try at homelabs-osaka, and I will back you. <3 thanks for your contributions

**Reply by capitanumihai** [reply]
> I checked stalwart, on the roadmap is "Mail client", lets be patient, I like what I see by the way, let me try at homelabs-osaka, and I will back you. <3 thanks for your contributions

Fast please. We need it! :D

**Comment by shahryarjb** [comment]
There is no new news for mail server?
Thank you in advance

**Reply by OSZII** [reply]
unfortunatly not...

**Comment by ebulku** [comment]
+1

**Comment by bst1n** [comment]
can't Plunk do the job ?

**Comment by Cinzya** [comment]
Stalwart has been added as service to Coolify, which you can now deploy on your server and use as a mail server. It does require some additional steps to make sure the proxy is not causing problems. So make sure to closely read the [Coolify Docs](https://coolify.io/docs/services/stalwart), after deploying Stalwart.
